pkg/ipcache/metadata.go

```diff
@@ -179,6 +179,9 @@ func (m *metadata) getLocked(prefix netip.Prefix) PrefixInfo {
 // Returns the CIDRs that were not yet processed, for example due to an
 // unexpected error while processing the identity updates for those CIDRs
 // The caller should attempt to retry injecting labels for those CIDRs.
+//
+// Note: do not call this function (except in tests)! Rather, calling
+// UpsertLabels() / UpsertMetadata() will trigger an update asynchronously.
 func (ipc *IPCache) InjectLabels(ctx context.Context, modifiedPrefixes []netip.Prefix) (remainingPrefixes []netip.Prefix, err error) {
```

Comment:
Incidentally we can probably hide this with a rename to lowercase this as well as  `TriggerLabelInjection()` to prevent such misuse.

One of these days we should probably rework the diagram as well, since there are more and more sources of information using this model. But I digress.

These suggestions are unrelated to the PR, can be submitted separately or as a separate commit.